src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs

```diff
@@ -3131,6 +3131,363 @@ public Task DisposeAsync()
+                  Operand: 
+                    IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: C, IsImplicit) (Syntax: 'this')
+            Next (Regular) Block[B4]
+        Block[B4] - Block
+            Predecessors: [B3]
+            Statements (1)
+                IAwaitOperation (OperationKind.Await, Type: System.Void, IsImplicit) (Syntax: 'this')
+                  Expression: 
+                    IInvocationOperation ( System.Threading.Tasks.Task C.DisposeAsync(params System.Int32[] extras)) (OperationKind.Invocation, Type: System.Threading.Tasks.Task, IsImplicit) (Syntax: 'this')
+                      Instance Receiver: 
+                        IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: C, IsImplicit) (Syntax: 'this')
+                      Arguments(1):
+                          IArgumentOperation (ArgumentKind.ParamArray, Matching Parameter: extras) (OperationKind.Argument, Type: null, IsImplicit) (Syntax: 'await using(this){}')
+                            IArrayCreationOperation (OperationKind.ArrayCreation, Type: System.Int32[], IsImplicit) (Syntax: 'await using(this){}')
+                              Dimension Sizes(1):
+                                  ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0, IsImplicit) (Syntax: 'await using(this){}')
+                              Initializer: 
+                                IArrayInitializerOperation (0 elements) (OperationKind.ArrayInitializer, Type: null, IsImplicit) (Syntax: 'await using(this){}')
+                                  Element Values(0)
+                            InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                            OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+            Next (Regular) Block[B5]
+        Block[B5] - Block
+            Predecessors: [B3] [B4]
+            Statements (0)
+            Next (StructuredExceptionHandling) Block[null]
+    }
+}
+Block[B6] - Exit
+    Predecessors: [B2]
+    Statements (0)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source + s_IAsyncEnumerable + s_ValueTask, expectedGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void UsingFlow_21()
+        {
+            string source = @"
+using System;
+using System.Threading.Tasks;
+public class C 
+{
+    public async Task M() 
+/*<bind>*/{
+        await using(this){}
+    }/*</bind>*/
+    
+    Task DisposeAsync(int a = 3, params int[] extras, bool b = false) => throw null;
```

Comment:
>   Task DisposeAsync(int a = 3, params int[] extras, bool b = false) => throw null; [](start = 1, length = 83)

I think we need to add more tests around an unconventional use of params/ParamArrayAttribute. For example:
1) What happens if we also add default value for ```extras``` parameter;
2) What happens if we consume a method like from #1 from metadata;
3) The last argument has params in metadata, but it is not an array and not optional; 
4) The last argument has params in metadata, but it is not an array, it is optional and doesn't have explicit default value; 
5) The last argument has params in metadata, but it is not an array, it is optional and has default value; 
6) The last argument has params in metadata, it is an array, it is optional and has default value.

We should also make sure that we have tests that are trying to emit code for scenarios like that, and verify what happens when emit succeeds.